manim/scene/scene.py

```diff
@@ -1006,15 +1004,32 @@ def play_internal(self, skip_rendering=False):
         # Closing the progress bar at the end of the play.
         self.time_progression.close()
 
+    def check_interactive_embed_is_valid(self):
+        if self.skip_animation_preview:
+            logger.warning(
+                "Disabling interactive embed as 'skip_animation_preview' is enabled",
+            )
+            return False
+        elif config["write_to_movie"]:
+            logger.warning("Disabling interactive embed as 'write_to_movie' is enabled")
+            return False
+        elif config["format"] == "png":
+            logger.warning("Disabling interactive embed as '--format png' is set")
+            return False
+        elif not self.renderer.window:
+            logger.warning("Disabling interactive embed as '--format png' is set")
+            return False
+        return True
+
     def interactive_embed(self):
         """
         Like embed(), but allows for screen interaction.
         """
-        if self.skip_animation_preview or config["write_to_movie"]:
+        if not self.check_interactive_embed_is_valid():
+            logger.info("interactive embed is disabled and will not be launched")
             return
 
         def ipython(shell, namespace):
-            import manim
```

Comment:
Oh, i forgot to comment this in the review: is this change safe?

Reply:
I don't remember doing that, must be a mistake so reverted